### DISCUSSION TITLE
MPC tune aborts with "Temperature error"

### DISCUSSION THREAD

**Discussion by nelis10000** [mriscoc/Ender3V2S1#784] [Q&A]
Hi,

I'm having problems tuning the MPC on my Ender 3S1 with an all-metal heating block.
This heating block has a 24V/40W heating element (measured cold resistance 14.8Ω) and a bi-metallic heat break.
Firmware used is Ender3S1-F1-UBL-MPC-20221222.bin.

- If I start an MPC tune **with** the silicone heat sock on the heat block, MPC tune fails.
  In the octoprint terminal, I can see "Temperature error" followed by a "MPC Autotune finished!" (all MPC variables are 0).
- If I run the MPC tune **without** the heat sock, it tunes just fine. 
- If I use the original printer's heat block, MPC tunes just fine with or without the silicone sock.

To troubleshoot this problem, I've attached an LED to the heater to get a visual indication of the heater PWM. 
What I see is this:
- cool to ambient - no PWM
- heat to 200C - PWM on 
- at 200C - PWM toggles on/off a few times, for a short duration
- after that, the PWM is just on at 100% and the temperature quickly increases
- MPC tune aborts with "Temperature error" because of the limit set in temperature.cpp (>15C overshoot).

(I did change the code to "_hotend.target + 30.0f_" but the result is the same, only 30C overshoot instead of 15C)

I compiled a new firmware, same as 20221222 above, but with the two debug blocks enabled (temperature.cpp, at lines 1088 and 1479).
This produces a lot of extra data on the Octoprint terminal, most of which I don't understand but this part looks odd to me:
Tune ok: _Recv: asymp_temp 895.12_
Tune fail: _Recv: asymp_temp 4294967295.42949672954294967295_

Attached is the terminal output for:
- new block, without heat sock
- new block, with heat sock
- original block, with heat sock (for comparison, the MPC values look similar to the new block without heat sock, don't they?)

Anyone who knows what's going on and (optionally but much appreciated) how to fix it?

Thanks,
Niels

[newblock-no_heatsock-dbg-OK-03.txt](https://github.com/mriscoc/Ender3V2S1/files/10951783/newblock-no_heatsock-dbg-OK-03.txt)
[newblock-heatsock-dbg-FAIL-03.txt](https://github.com/mriscoc/Ender3V2S1/files/10951785/newblock-heatsock-dbg-FAIL-03.txt)
[origblock-heatsock-dbg-OK-03.txt](https://github.com/mriscoc/Ender3V2S1/files/10951817/origblock-heatsock-dbg-OK-03.txt)

**Comment by mriscoc** [comment]
Please try with latest firmware version. Also try with the latest Marlin bugfix. Report MPC issues in Marlin's repository.

**Comment by nelis10000** [comment]
Hi mriscoc,

I wasn't sure where to post this issue. Thanks for the pointers!
